2021/day11/day11.java

```diff
@@ -0,0 +1,141 @@
+            for (int k = 1; k < octoGrid.get(j).size() - 1; k++) {
+                octoGrid.get(j).set(k, octoGrid.get(j).get(k) + 1);
+            }
+        }
+    }
+
+    public static void increaseOne(ArrayList<ArrayList<Integer>> octoGrid,
+            int x, int y) {
+        if (octoGrid.get(y).get(x) != 0) {
+            octoGrid.get(y).set(x, octoGrid.get(y).get(x) + 1);
+        }
+    }
+
+    public static void processFlashes(ArrayList<ArrayList<Integer>> octoGrid) {
+        for (int j = 1; j < octoGrid.size() - 1; j++) {
+            for (int k = 1; k < octoGrid.get(j).size() - 1; k++) {
+                if (octoGrid.get(j).get(k) > 9) {
+                    octoGrid.get(j).set(k, 0);
+                    increaseOne(octoGrid, k - 1, j);
+                    increaseOne(octoGrid, k + 1, j);
+                    increaseOne(octoGrid, k + 1, j - 1);
+                    increaseOne(octoGrid, k + 1, j + 1);
+                    increaseOne(octoGrid, k - 1, j - 1);
+                    increaseOne(octoGrid, k - 1, j + 1);
+                    increaseOne(octoGrid, k, j - 1);
+                    increaseOne(octoGrid, k, j + 1);
+                }
+            }
+        }
+    }
+
+    public static boolean isFlashReady(ArrayList<ArrayList<Integer>> octoGrid) {
+        for (int j = 1; j < octoGrid.size() - 1; j++) {
+            for (int k = 1; k < octoGrid.get(j).size() - 1; k++) {
+                if (octoGrid.get(j).get(k) > 9) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static int countFlashes(ArrayList<ArrayList<Integer>> octoGrid) {
+        int counter = 0;
+        for (int j = 1; j < octoGrid.size() - 1; j++) {
+            for (int k = 1; k < octoGrid.get(j).size() - 1; k++) {
+                if (octoGrid.get(j).get(k) == 0)
+                    counter++;
+            }
+        }
+        return counter;
+    }
+
+    public static boolean areAllFlashing(ArrayList<ArrayList<Integer>> octoGrid) {
+        for (int j = 1; j < octoGrid.size() - 1; j++) {
+            for (int k = 1; k < octoGrid.get(j).size() - 1; k++) {
+                if (octoGrid.get(j).get(k) != 0)
+                    return false;
+            }
+        }
+        return true;
+    }
+
+    public static int processStep(ArrayList<ArrayList<Integer>> octoGrid) {
+        int flashCounter = 0;
+        increaseAll(octoGrid);
+        while (isFlashReady(octoGrid)) {
+            processFlashes(octoGrid);
+        }
+        flashCounter += countFlashes(octoGrid);
+        return flashCounter;
+    }
+
+    public static int countAllFlashes(int step) {
+        ArrayList<ArrayList<Integer>> octoGrid = readInput("data.txt");
+        int flashes = 0;
+        for (int i = 0; i < step; i++) {
+            flashes += processStep(octoGrid);
+        }
+        return flashes;
+    }
+
+    public static int findSyncStep(int step) {
```

Comment:
I would change `step` into `maxStep`, as this is the maximum step you're willing to check